docs/source/conf.py

```diff
@@ -112,8 +112,8 @@
         "typing.Type",
         "typing.Set",
         "kedro.config.config.ConfigLoader",
-        "kedro.io.core.AbstractDataSet",
-        "kedro.io.core.AbstractVersionedDataSet",
+        "kedro.io.core.AbstractDataset",
+        "kedro.io.core.AbstractVersionedDataset",
```

Comment:
```suggestion
        "kedro.io.core.AbstractDataSet",
        "kedro.io.core.AbstractVersionedDataSet",
```
Shouldn't be necessary.